ansible/roles/operator-pipeline/templates/openshift/tasks/preflight.yml

```diff
@@ -19,20 +19,57 @@ spec:
       description: The Operator index image which includes the bundle
     - name: bundle_image
       description: The Operator bundle image under test
+    - name: service_account
+      description: The service account for the preflight config
+      default: "default"
+    - name: namespace
+      description: The namespace for the preflight config
+      default: "default"
   results:
     - name: log_output_file
     - name: result_output_file
     - name: artifacts_output_dir
   workspaces:
+    - name: output
+      description: A workspace to keep track of the task while PipelineRun.
     - name: kubeconfig
       description: A kubernetes config for the cluster used in preflight testing
+    - name: credentials
+      description: Docker config for retrieving the bundle image
+      optional: true
   steps:
     - name: check-operator
       image: $(params.base_image)
+      workingDir: $(workspaces.output.path)
+      env:
+        - name: PFLT_INDEXIMAGE
+          value: $(params.bundle_index_image)
+        - name: PFLT_SERVICEACCOUNT
+          value: $(params.service_account)
+        - name: PFLT_NAMESPACE
+          value: $(params.namespace)
+        - name: PFLT_LOGFILE
+          value: preflight.log
+        - name: PFLT_ARTIFACTS
+          value: artifacts
+        - name: KUBECONFIG
+          value: $(workspaces.kubeconfig.path)/kubeconfig
       script: |
         #! /usr/bin/env bash
         set -xe
 
+        if [[ "$(workspaces.credentials.bound)" == "true" ]]; then
+          # Combine the default credentials
+          # with those found in the workspace to maintain access to the internal
+          # registry.
```

Comment:
This is minor but you could probably arrange these comments together a little better.